---
title: "DWQE review questions"
output: pdf_document
header-includes: \usepackage{graphicx}
fontsize: 10pt
geometry: margin=1in
---

## Mike Alfaro
### Questions from past exams   

**2010**  
\includegraphics[width=14cm]{dwqe_figs/phylo_traits.png}


Table 1 represents log-transformed raw measures of beak depth and wing length for six species of galapagos finches. Consider the table and phylogeny above and answer the following questions.   

A)  If regression analysis of the tip data in Table 1 showed a significant positive relationship would you be justified in concluding that an evolutionary association exists between the traits? Why or why not?   

B) Calculate standardized independent contrasts for nodes a-e. Show your work!.   

C) What are contrasts and why are they used in comparative biology? D) What model of character evolution is assumed to underlie the distribution of character values on the tips of the tree? 

---------

Modern cetaceans (whales and dolphins) comprise 84 species and evolved approximately 35 million years ago.  Consider the hypothesis that their diversity in species richness and morphology is due to an ecological adaptive radiation following the extinction of the archaeocetes (a paraphyletic assemblage of stem cetaceans that went extinct around the time of origin of crown cetaceans).    

A)  Your lab has recently produced an exhaustive time-calibrated phylogeny for all of the modern species. Explain how you could use this tree to test the hypothesis that species richness in cetaceans resulted from ecological adaptive radiation. Describe at least two comparative methods you might use and discuss the results you would expect from these analyses that would confirm or reject the hypothesis.   

B)  The dolphins, a subclade of cetaceans with 32 species, evolved approximately 10 million years ago. Explain how the rapid evolution of this species-rich young clade would affect the analyses you discussed in B). 

C)  Your hard-working graduate students have also assembled a detailed data set of skull measures for all cetacean species.  Explain how you could use the data set in conjunction with the phylogeny to test whether morphological diversity is consistent with the hypothesis of ecological adaptive radiation. What patterns do you expect to see in the data if the hypothesis is true?  What are some of the alternative hypotheses?  How might you evaluate them with your data set? 

D)  How would ecological limits on species richness or character evolution affect your tests of the ecological adaptive radiation hypothesis?  What results would you expect to see if cetaceans had reached these limits early in their history?

---------

\includegraphics[width=9cm]{dwqe_figs/SI_index.png}

Suction index is is a measure of the capacity of fishes to generate suction by explosive expansion of the skull and buccal cavity.  This index is determined by five morphological traits: The cross-sectional area (CSA) of the expaxial muscle, the mechanical advantage (ratio of in-lever to out-lever) of the epaxial, gape width, and length of the buccal cavity.  

A)  Describe three different morphologies that might be found in fish with high suction indices?  

B)  You find that a clade of catfishes have higher disparity than pufferfishes in these five traits.  Can you also conclude that diversity in suction index is also higher within catfishes? Why or why not? 

C)  Do you expect morphological diversity of the five traits or diversity in suction index to be a better predictor of trophic diversity? Explain your answer. 

D)  How could you assess whether the epaxial CSA + in-lever and out-lever was a distinct evolutionary module from buccal length + gape width in catfishes?

E)  How are functional modules influenced by developmental modules? What is one way that functional modules can feed back upon developmental modules? What are genetic modules and how do they interact with developmental modules?

---------

1. Your have been awarded a grant to test whether the diversity of anglerfishes (~300 species with a conspicuously high degree of morphological disparity) has been primarily driven by **ecological adaptive radiation following invasion of deep water habitats**. Your preliminary analyses reveal at least three separate lineages that have independently invaded deep waters. You have ammased high quality dissue samples and collected vouchers for all species.    

A)  Explain the phylogenomic technique you will use to construct a phylogeny based upon thousands of genetic markers. What advantages does your approach hold over traditional single-gene PCR and sequencing.   

B)  Discuss gene-tree/species-tree, concatenation, and short cut GT-ST methods for inferring phylogeny. Which method or methods will you use and why?    

C)  Which criteria of ecological adaptive radiation are you best equipped to test given your anglerfish samples? What kep predictions does this framework make about the tempo of anglerfish species diversification and morphological evolution?   

D)  Describe the kinds of morpho traits you will measure to best test the idea that ecological diversification underlies anglerfish radiations in deep waters?  

E)  Briefly describe a technique or method you will use to test hypotheses regarding the tempo of anglerfish **species diversification**. What results would you expect to see if your central hypothesis is correct?    

F)  Briefly describe a technique or method you will use to test hypotheses regarding the tempo of anglerfish **morphological evolution**. What results would you expect to see if your central hypothesis is correct?    

---------

2. How do biologists study species diversifiation? Discuss some central questions in diversification using examples from readings, research or the literature. What methods can be used to test hypotheses in this area? In your answer, include some consideration of at least three of these areas:    

A)  Diversity dependent vs. independent models;    
B)  Challenge of estimating extinction from molecular phylogenies;   
C)  Drief descriptions of at least three models of species diversification;  
D)  Hypothesis testing via model selection;   
E)  Approaches to studying patterns of diversification in the fossil record.   

---------

3. Jump in:   

A)  What are independent contrasts and why are they used in comparitive biology? What central assumptions of standard statistical methods is violated when performing interspecific comparisoins of trait values? What is the underlying model of trait evolution for independent contrast? Discuss three models of trait evolution covered in class that differ from the model underlying independent contrasts. Would contrasts be an appropriate method for analyzing traits that have evolved under these models? Why or why not?    

B)  Draw a fully resolved, bifurcating phylogeny for three species which shows hyraxes as the sister group to elephants + manatees. Assume that the age of the split between hyraxes and elephants + manatees is 50 million years and that the split between elephants and manatees happened 25 MYA.  Further assume that the log transformed masses of the species is as follows: Hyrax 1.5, Elephant 8.0, Manatee 6.0.  Calculate the standardized contrast for each internal node. 

\clearpage

## Bob Wayne

### Lecture 3- Scope of molecular ecology

1. Individual relatedness and population pedigrees:  
  -  Reproductive success, skewness and mating system
  -  Cooperation and social hierarchies
  -  Inbreeding and fitness 

**Within groups/demes**  
-  Gray wolves- population pedigree, mating system, inbreeding, reproductive success, and kinship. Disassortative mating between white and black wolves. Most reproductive pairs were also the dominant individuals.  Pack structure was varied. Showed that inbreeding is rare.  Dispersal happened by males moving     

2.  Population variation, demography, and gene flow.  
**Genetic variation within populations**  
-  What is the effective size of the population? What is demographic history? 
-  Is there ongoing migration between populations? In mountain lions across Santa Monica mountains, number of migrants is high - but too high to explain Fst (migrants moving but not really mating into the new populations)
-  What is the best predictor of genetic differentiation? Isolation by distance; vegetation analysis, etc. 

3.  Phylogeography  

-  Effect of isolation barriers- topography can explain genetic hierarchies. Deer mouse populations across the continental US. This leads to the correspondence of geography and phylogenetic patterns.    

4.  Genetic basis of adaptation  

Selective sweeps leave a pattern in the genome- reduced variance around selected loci. Identify candidate loci that might be under selection- use things like Fst and other approaches. 

Example: Dymeclin, which is involved in bone development and causes some syndrome- skeletal development, barrel shaped chest, knock knees, low joint mobility.  

Follow up scans with exon capture or some other functional approach

5.  Gene expression and epigenetics

-  RNA seq- stress factors can affect pattern of gene expression. Can then try to knock out th gene or introduce edits, for example now with CRISPR.   

6.  Metagenomics and eDNA

7.  Non invasive monitoring  

8.  Ancient DNA  



### Questions from past exams   

What are some of the most significant problems in conservation genetics? What are some of the definitions of evolutionarily significant units? Describe how some of these questions can be addressed using molecular genetic techniques and illustrate with examples from class, your textbook, or readings.

---------

What are some of the defining questions in molecular ecology at the individual, group, population and species level? What is a phylogeographic pattern and how might it be useful for understanding population history and adaptation? Provide examples from class, your textbook, or readings.

---------

What are some of the major goals of genomic science? Discuss applications of genomic science to systematics and to the search for genes under selection and that underlie phenotypic traits. With regard to finding genes under selection, contrast comparative genomic and population genetic approaches.

---------

Discuss two or three aims of genomics that have strong evolutionary components. Provide examples  

---------

The relative new field of molecular ecology has applications to behavior, population genetics, evolutionary history and conservation biology. Discuss the field of molecular ecology with regard to each of these disciplines and provide examples from class or the readings   

---------

A)  What is a phylogeographic pattern and why do such patterns occur? provide an example   

B)  Discuss several evolutionary questions that can be addressed by analysis of mtDNA phylogenies below the species level 

C)  How can a mtDNA phylogeny be used to test a model of sympatric speciation? Provide an example.    

\clearpage

## Kirk Lohmueller   

### Lecture 3- principles of natural selection    
**What happens to a selected allele in a finite population**?   
Note that selection on a particular allele is deterministic in the absence of other evolutionary forces.   

*First, let's consider what happens to an adaptive allele in an infinite population*   
The allele should go to fixation. 
Fully dominant allele will rapidly rise, but it will be slow to reach fixation- a lot of the recessive alleles will be left around because the heterozygotes show the dominant phenotype.  
Additive alleles will rapidly rise and go closer to fixation, because homozygotes have a higher fitness than the hets.   
Recessive alleles will be slow to rise, since individuals won't start showing the adaptive phenotype for a while.    

If there is heterozygote advantage, the alleles will be at some intermediate frequencies.  
Equilibrium allele frequency does not depend on initial frequency of the allele:    


$$p_{A} = \frac{s_{BB}}{s_{AA}+s_{BB}}$$

**In a finite population, drift and selection act on alleles at the same time**  
*p(Fixation) ~ population size and strength of selection*   

$$u(S,N) = \frac{1-e^{-2s}}{1-e^{-4Ns}}$$

Basic rule is that p(Fixation, neutral) = initial frequency = 1/2N. Higher for advantageous alleles, lower for deleterious alleles. Many adaptive alleles are lost, some deleterious alleles reach fixation. Strongly deleterious alleles are lost quickly.    

The adaptive alleles that do make it to fixation do so much faster in finite populations than in infinite populations- rate is higher than $1/2N$.   



But for some reason, p(Fixation of strongly selected alleles) is insensitive to $N$ in large, infinite population- the $p(Fix) = 2s$. The intuition behind this is that early on when the frequency of the allele is low and it more or lless follows mendellian genetics.  

### Lecture 4- Molecular pop gen of natural selection   

**What do neutral models predict about species divergence?**    

Neutral theory assumes that advantageous mutations may occur, but at such a low frequency that they may be safely neglected. If the mutation rate per gamete is $v_{T}$ and a fraction $f_{0}$ are neutral (the other are strongly deleterious and lost from the population), then then the rate of substitution $k_{g}$ follows

$$k_{g} = (2N*v_{T}*f_{0})*(\frac{1}{2N}) = v_{T}*f_{0}$$

So that the rate of substitution is not a function of population size- just a function of how often neutral mutations come up. The biological intuition behind this is that more mutations may come up in large population, but given that coal time will be longer they are more likely to be lost; inverse in small populations.   

**Selective neutrality** is the idea that the neutral mutation rate can change with locus- if a protein needs to be spot on the same, then a larger fraction of new mutations will be deleterious- which results in a smaller $f_{0}$ and a lower rate of substitution at that site.    

**Detecting selection using divergence**   

*What happens when there is positive selection on non-synonymous mutations?*   
-Assume that synonymous mutations are neutral- synonymous substitutions occur at a rate of the neutral mutation rate. Let's call this $d_{s}$  

- Non-synonymous mutations can be under selection:  
- *if* they happen to be neutral, their substitution rate should be the same as $d_{S}$   
- if they are under *negative selection*, they should accumulate substitutions at a slower rate than neutral ($d_{N} < d_{S}$)
- if they are under *positive selection*, they should accumulate substitutions at a slower rate than neutral ($d_{N} > d_{S}$)

Evidence suggests that there is strong purifying/negative selection- $d_{N}/d_{S}$ tends to be much lower than 1.  
Nielsen found 700 genes with $d_{N}/d_{S} > 1$- but none significant. Not clear how to interpret it- often times these will be just reported as a list or as a list of GOs.   
FOXP2 thought to drive language in humans- positive selection hypothesized.  $d_{N}/d_{S}$ tends to be much lower in non-human mammals, but really much higher in humans. 

$d_{N}/d_{S} > 1$ is strong evidence for positive selection, but it is a weak tests- there will tend to be many false negatives.   

**Detecting positive selection from polymorphisms**    

Classic selective sweep model assumes that a strongly selected mutation arises and quickly reaches fixation. This leaves some footprints:    

-  Reduction in diversity;   
-  Skew towards rare alleles;  
-  Skew towards high-frequency derived alleles;   
-  Large differences in allele frequncies between populations (if there is local adaptation). 

Case study: strong sweep pattern found around leg length gene in dachshunds; very low Tajima's $D$ (lots of rare-frequency alleles) in FOXP2 indicates recent strong selection.   

**Haplotype test**   
If there is a strong selective sweep in the past, we should see long haplotypes around high frequency alleles. If an allele reached high frequency through drift there probably won't be a long haplotype around it.   

Coop et al. showed that some alleles are on long haplotypes in all non-African human populations they sampled; other derived allleles are on long haplotypes in only some of the non-African populations they sampled.    

**Differences in allele frequency** (can be measured as $F_{ST}$, higher = bigger difference) can be driven by selection in one population relative to another.   

**Ancient DNA** can tell us about selection- we can compare allele frequency between ancient populations and modern populations; we've found out recently that a lot of alleles have been under selection in modern Europeans relative to ancient remains.    

**Difficulties with selection scans**   
A lot of the signatures are difficult to separate from signatures of demography.   
Skew to rare alleles- caused by selection; caused by rapid growth of population.    
Reduction in diversity- caused by sweep; caused by bottlenecks.   
High frequency long haplotypes- caused by sweep; caused by bottlenecks.   

**How to work around this**   

1. Outlier approach- just pick the most extreme loci. But even in the absence of any selection there will be *some* "most extreme"!   

2. Model-based approach- fit neutral model to genome wide data, and look for loci that are unusual under this complex model.   

Tests of selection are sensitive at different time scales- so it would make sense that we identify different loci as being under selection in different tests.  

Two-dimensional SFS- compares allele frequency in two populations. Cool! Found that EPAS1 hypoxia factor selected in Tibetian populations relative to Han Chinese. But this gene not implicated in hypoxia adaptation in Ethiopean populations. Curiously, the EPAS 1 in Tibetian populations seems to come from Denisovans.  

**So far** we have considered hard sweeps/hard selection- but there are polygennic models of selection which are probably more realistic. Multiple loci control trait of interest, seleciton acts by shifting frequncy of many alleles; no loci become fixed.   

### Lecture 5- what determines genetic variation in species?   

1. Genetic drift:  

-  largely determined by population size;  
-  bottlenecks and population structure may be important;   
-  life history traits may be important.   

2. Selection:  

-  Selection can reduce variation at selected sites;  
-  Also reduces variation at linked sites.  

Leffler et al's work on genetic diversity suggested that the variation in census size is much higher than variation in genetic diversity- not clear why. Larger population species tend to show more diversity, in agreement with theory, but the **quantitative** pattern does not match up.  

*Can natural selection be causing similar levels of diversity among species?*   

-  Sites directly under selection can have low diversity due to purifying selection.  
-  Sites linked to others under selection can have low diversity due to sweeps eliminating diversity, and because of background selection (diversity loss at sites linked to new deleterious mutations).   

**Evidence for selection affecting linked neutral sites**   

A recurring theme here seems to be that a correlation between recombination rate and neutral neucleotide diversity is evidence for natural selection:  

\includegraphics[width=14cm]{dwqe_figs/recomb_selection.png}

The idea here is that at low rates of recombination there is lots of linkage, and little diversity. But at regions of high recombination rates where a lot of alleles might be essentially independent, the diversity is much higher.    

In *Drosophila*, there is documentation of a drop in neutral diversity surrounding non-synonymous substitutions.  

In humans, the correlation between recombination rate and neutral diversity exists, and is stronger in genic regions than in non-genic regions.   

But background selection (purifying selection) may also drive these patterns- not sure what is doing the driving.  

Diversity at neutral sites linked with selected sites is lower in species with large populations than in species with small populations.  This might explain part of the fact that the quantitative relationship between N and genetic diversity is not held up.    

### Lecture 6- fitness effects of mutations and evolutionary quant gen   

\includegraphics[width=14cm]{dwqe_figs/fitness_effects.png}

We can think about fitness effects under **Fisher's geometric model**  

*If the population is not at optimal fitness*, the liklihood that a *small effect mutation* is beneficial is 50%. But *large effect mutations* are unlikely to be beneficial. Beneficial mutations are predicted to have smaller effects as population moves closer to the optimum.  

**Fischer/Wright's Fitness Landscape model** proposes that selection causes populations to "walk" up to peaks in the adaptive landscape- usually up to the closest peak. For a mutation to be beneficial, it should push up the population further towards the peak.  

**Molecular approaches to studying fitness effects**   

*Experimental approaches*   

-  Mutation-Accumulation experiments:   
An ancestral line is split into several new lineages, which are allowed to evolve separately. At the end, some sort of competition experiment is done on the same environment to check which lineage has a higher fitness- the "OG" or the derived lineages that have been mutating.  

In viruses, some evidence that most mutations are lethal, many are nearly neutral.  
In yeast, some evidence that most mutations are nearly neutral, some are lethal.   

-  Directed mutagenesis approach:  
Hietpas et al. examined every possible mutation in a 9 Amino Acid stretch in a Heat Shock Protein. For some positions, most mutations were neutral; for other positions, most positions were lethal or at least strongly deleterious. 

If there is a shift in the environment, (for example, Lenski exposing bacteria to a new medium), Fisher's geometric model would predict that early mutations should increase population fitness, and the rate of fitness increase should decrease with time.  

**In summary** we have found many weakly or strongly deleterious mutations, rare advantageous mutations (in constant environment- more if env changing). But if there are small effect mutations their signal is unlikely to come through in experiments.   

*Indirect approaches*  
 
**McDonald-Kreitman tests**   
Tests if new SNPs come out at faster rates than they go to fixation.   

**Site frequency spectrum**  
If there is purifying selection, there are a lot more rare alleles-  makes sense, since they never really rise from rarity.  

The way to do this is to make an SFS for synonymous and non-synonymous sites, generate a demographic model based on the synonymous sites, and estimate selection from the difference between the Non-Synonymous SFS and that expected by the model.   

The Gamma distribution seems to emperically do a good job of predicting SFS at non-synonymous sites.  

**Boyko et al 2008:**   

-  70-80% of new non-synonymous mutations were weakly to strongly deleterious. 
-  Most mutations that reach fixation are neutral.  
-  Models with some purifing selection are better than those without.   

--------------

**How does genotype influence phenotype?**    

For an individual, we can write that:   
$$Phenotype = Mean + Genetic\ Effects + Environmental\ Effects$$ 
Or for a population
$$Phenotype\ Variance = Genetypic\ Variancce + Environmental\ Variance$$ 

**Heritability**- blechhhhhhhh    

Broad sense heritability: 
$$H^{2} = \frac{Genetic\ Variance}{Phenotypic\ Variance}$$
- Note that this is a property of a population; not a trait!  

What is genetic variance:  
\includegraphics[width=14cm]{dwqe_figs/genetic_variance.png}

Narrow sense heritability:  
$$H^{2} = \frac{Additive\ Variance}{Phenotypic\ Variance}$$
-  Summarizes the amount of phenotypic variance attributable to additive effects- discludes genetic interactions and dominance.  
-  Evol biologists care because this is what determines response to selection.    

-------------------

**GWAS** results from humans- a summary:   

-  most GWAS hits are in non-coding regions;  
-  most associated variants were common - but the tests were designed to find common variants to not sure how biologically meaningful this is;  
-  little evidence for epistasis- but again, tests don't have enough power to detect this always;  
-  single-SNP effects are quite weak;   
-  GWAS work better for some traits (e.g autoimmune) than others (e.g. psychiatric disease)


**Missing heritability** - read up on this.    

\clearpage 

### Questions from past exams   

**2014**   

You are interested in studying population genomics of domesticated rabbits. To do this you sequence the genomes of six domesticated rabbits and six wild rabbits from the same ancestral population from which rabbits were domesticated. From this sequencing effort, you identify many SNPs across the genomes of these animals

A)  You divide the genome into 100kb windows and look at the numbers of SNPs per window in the domesticated rabbits. Supposing all of the windows are neutral (i.e. no selection), why would the number of SNPs vary from window to window?

B)  Take a look at the sliding window plot for part of the genome. It shows the heterozygosity and the $F_{ST}$ between wild and domesticated rabbits. What do you observe? Outline your interepretations and hypotheses, and give two strategies to test your hypothesis.   

\includegraphics[width=17cm]{dwqe_figs/genome_scan.png}

C) You are interested in learning about the demographic history of domesticated rabbits. Describe a computational strategy to analyze the gentic variation data to estimate whether there was a bottleneck associated with domestication. Make sure you address these things:
  - What general computational strategy would you use for inference  
  - What population genetic model would you use?   
  - How might you summarize the observed genetic data?   
  - What are some pros and cons?    
  
D) During the course of the analysis, you notice that there is a greater percentage of SNPs in UTR and conserved non-coding regions show high $F_{ST}$ values compared to random intergenic SNPs. Briefly (1-2 sentences) describe one possible evolutionary process that could cause this pattern. 

E) You are interested in finding the effects of artificial selection that occurred during domestication. Let's further suppose that you have estimated that rabbits were domesticated within the last 1400 years. You need to decide how to do your selection scan. You have two strategies in mind. First is to use the $d_{N}/d_{S}$ test to compare domesticated rabbits to mice (which were diverged at least 50 MYA). The secnd is to use an $F_{ST}$ based analysis comparing the differences in allele frequencies between wild and domesticated rabbits. Which approach would be more powerful, and why? Provide two reasons.  

F) Place a mark on the figure below to indicate the coalescent time between the two chromosomes. 

\includegraphics[width=9cm]{dwqe_figs/coal_time_tree.png}

---------

A)  Assume that a population is well adapted to its environment. Would most new mutations be expected to be advantageous or deleterious? Give two lines of evidence supporting this conclusion.   
B)  What effect does negative natural selection have on the site frequency spectrum? Asked another way, how does the SFS for deleterious mutations compare to the frequncy spectrum of neutral mutations?    

C)  Give one other evolutionary force (i.e. selective or demographic) that may give the same pattern in the SFS as that produced by negative selection.

D)  Mutation accumulation lines can be used to study the fitness effects of new mutations. Give one disadvantage of this approach to study the fitness of mutations.    

E)  What does the Neutral theory of molecular evolution say about deleterious mutations? How does that differ from the nearly neutral theory?    

F)  Are most new mutations that occur in coding regions of human genomes deleterious or adaptive? Give at least one line of evidence (some pattern seen in genetic data) that supports this view.   

---------

Models of genetic drift depend on population size N. Describe how N is important for:   

A)  Determining rate of coalesence.

B)  Determining the probability of gene tree/species tree discordance 

C)  The amount of neutral genetic variation found in a sample from the population   

D)  Describe how population size affects the rate of substitution for neutral alleles. 

E)  Assume that a neutral mutation has a frequency of 10 percent in a population. How does the population size affect the probability that the mutant allele becomes fixed in the population?   

F)  Assume that a positively selected mutation (s = 0.05) just arose in the population. How does the population size affect the probability that the positively selected allele becomes fixed?   

