---
title: "Evolution 200A"
author: "Gaurav Kandlikar"
output: html_document:
  toc: true
---

## Bob Wayne

### Admin
"Connecting genes to phenotypes"-  
Using capture arrays + resequencing to assess genetic basis of phenotypic variation  
Challenging cell lines from wild individuals/ gene editing  
Long term demography from genetic variation  

**Paper critiques**  
First paragraph- [Link to Nature's guide](http://faculty.washigton.edu/kate1/ESS_418/Class_Schedule_files/HandoutDay04.pdf)  

Then, fundamental problems- what's not so great about it from basic question perspective  

How to test the new ideas presented in the paper  

More in `wayne/critique_notes.doc`

### Intro to evolution  
The concept of evolution has changed since Darwin- matured into thinking about genotypic changes across time.  
A variety of sources of evidence have been used to develop the theories of evolution. The fact of evolution- that there was descent with modification- long predates Darwin. 

Four premises led Darwin to his theory of evolution-   
1. More individuals are born than can possibly survive;   
2. There will be a struggle for existence;     
3. There is some phenotypic variation among the individuals;  
4. So, the phenotypes that are better at reproducing will grow in proportion (i.e. evolution happens).  

At the time of Darwin, thinking was very typological- platonic ideals of species, with mistakes, not variation. Much of OOS is focused on anagenisis, rather than speciation per se.  

At the time of Darwin, there were a few key problems of 
1) No mechanism of inheritance;  
2) Gaps in geological records;  
3) Origin of complex structures.   

Post-Darwinian advances have included Mendel's work on particulate inheritence; Fischer, Mayr etc's pushing of theory forward; Motoo Kimura's neutral theory; modern discovery of abundant genetic variation.  

**Persistent problems**  
1. Evolution of sexual reproduction (two-fold cost);  
2. Imporatance of selection vs natural processes (selection leaves signal across genome, and many genes now seem to be under selection);  
3. Development (Evo-Devo)- how does substantial phenotypic change come about;  
4. Speciation;  
5. Unit of selection.  


------------
### 1 October 2015  

**Darwin's lines of evidence**  
1. Direct observation;  
2. Comparitive anatomy;  
3. Embryology/development;  
4. Biogeography;  
5. Fossil record;
6. Taxonomy.  

*Direct observation*  
- Peppered moths- a good example of a well studied system. Observational, naturally replicated experiments, replicated over time, etc.  
- Pesticide resistance/DDT.  

*Comparitive anatomy/Embryology and development*  
- Artificial selection in rock pigeons and dogs.  
- Phenotypic diversity in dogs may be generated by shifts in where allometric development is truncated (juvenilization/neoteny). Use of genomics in understanding canid evolution- cool story here.  
    Highest mtDNA diversity in East Asia- suggests the most ancient origin there. Domestication happened before agriculture - unique to dogs. Functional grouping in dogs has genetic basis, except "toy dogs" (vonHoldt 2010). This makes sense, no? Not sure why Wayne & co. did not expect this result. Genome-wide studies reveal information about bottlenecks, dating of origins, etc. Ancient DNA work has been done in dogs- kind of cool that we can see where modern dogs fall out relative to the ancient DNA (derived from skeletons). Wolves that are directly ancestral to modern dogs are probably not present today.  

*Biogeography*  
Continents tend to have organisms with similarities that likely arose through descent with modification. (Divergence)    
Continents also tend to have similar functional groups but with different body plans (eg herbivores). (Convergence)  
New/old world vultures are convergent- stork- and hawk-like ancestry, respectively.  
Island radiations were another key example of descent with modification- tenrecs & lemurs on Madagascar, finches on Galapagos and island honeycreepers in Hawaii are three classic examples.   
Clines- a third type of biogeographic evidence (change in phenotype with geography)  
Ring species later became a key model system.  

*Fossil record*  
- Fossils share phenotypes due to DWM;  
- There are differences between modern species and fossils- more DWM;  
- Transitional fossils show that things have changed according to environment;  
- Extinction necessitates a mechanism for generation of more diversity.  

Evolution of the mamalian middle ear- movement of bone from mandible to middle ear  

**The history of life is "bushy"**   
**Evolution represents diversification, not orthogenesis**. Distributions tend to become larger, variance changes with mean. 

*Embryology and Development*  
Darwin used this quite seriously. For a while, evolutionary biologists were all about recapitulation- embryos go through all previous life forms.  

*Taxonomy*  
Shared characteristics between organisms can only be explaiend by DWM.  

*Comparitive anatomy*  
Work by Cuvier showed that some structures were homologous- derived from shared ancestry; others, like bird and bat wings, were analogous (despite the individual structures, like the separate bones, being homologous).  
Vestigial structures are another example of why comparitive anatomy provided useful evidence about evolution. Some horses end up developing an extra toe because of incomplete gene suppression.  
Selection is constrained by contingency.  

----------

### 5 October 2015   

**Molecular Ecology**  
Adaptive and deleterious variation is a key frontier for mol ecol- how does selection change gene frequencies?

The field has a history in studying neutral markers to infer demographic processes- but there is now a shift to ecologically important genes- eg. how does selection drive local adaptation?  
Mol Ecology has been revolutionary in some fields- it used to be thought that birds are primarily monogamous, but genetic tests of egg clutches revealed the impacts of all sorts of extra-pair copulation, etc.  

Questions  
A) Individual relatedness and population pedigrees;  
B) Population variation, demography and gene flow;  
C) Phylogeography;  
D) Genetic basis of adaptation;  
E) Gene expression;  
F) ;  
G) ;  
H) Ancient DNA.  

*There are many potential problems*-  
Locus specific effects based on type of data (SNPs vs Allozymes vs Microsats, ...)  
Analytical models- assumptions are often broken  
Sampling issues- sample size and biases  
Methodological issues- cost? appropriate technique?  

mtDNA often used for species ID (Barcoding), assessing level of genetic polymorphism within and between populations.  
Microsats often used for individual ID, relatedness estimates, reconstructing pedigrees, census estimates and Ne, level of polymorphism within and between populations. Microsats problematic beyond intraspecific because of convergence in microsat #.  


**Individual based studies**   
*within group*:  pedigree, mating systems, inbreeding, reproductive success, parentage, and kinship  
Pedigree made for full population of Yellowstone wolves- basis for anything we can say about heritability. Dominance of genes determined this way- black coat was dominant in the wolf case. Ancestry of genes analyzed- black allele came from dogs and confers immunity against distemper. Disassortative mating between black and gray wolves revealed. Inbreeding was shown to be really rare and only occurred in extreme situations.   

*between group*:  inter-group relatedness, dispersal, inter-group interactions   
Can provide insight into conservation biology- population structure analysis of SaMo Mtn Lions showed that many individuals were killed due to density of individuals, not necessarily due to scarcity.  

**Population based studies**  
Genetic variation within population tells us sometihng about effective population size (Ne)- what is the impact of drift?  Demographic history of a population can also be estimated this way- bottlenecks? population expansions? selection?   
Genetic differentiation among populations- are there migrants from one population to another? Do migrants reproduce in new population? Allele differentiation, landscape genetics  
Rare alleles are very common- very similar to SAD in forests. Over time, stretches of homozygosity are broken up- this is also a function of effective size, so demographic history reconstructed this way (?) ([PSMC](http://lh3.github.io/2014/07/20/alternatives-to-psmc/)). 

**Landscape Genetics**  
Model genetic patterns across landscape as f(environmental parameters). Isolation by Distance is the crudest version of this- more distant populations have fewer migrants, so more differentiation with distance. But in some cases (e.g. migratory animals like wolves) geographical distance is less informative than environmetal dissimilarity (e.g. vegetation type). Genetic structure of wolves can be mapped onto vegetation type too. We can synthesize this approach with, MaxEnt to model the distribution of different ecotypes within a species. 

**Phylogeography**  
Landscape can structure genetics in a heirarchical way- this allows us to map distributions onto evolutionary histories of lineages. EG turtle beaches have distinct mt haplotypes- turtles are phylopatric.  

**Adaptive variation**  
Started off with MHC and neutral proxies. Modern studies usee genome wide studies, genome sequencing, and gene expression/epigenetics. When making chips of SNPs, we need to be careful about the population used for SNP discovery- what if there is a bias in the reference population? 

Selective sweep- variation is reduced at a given site. Determine candidate SNPs under selection by looking for SNPs with really high Fst values and something about homozygosity - figure out the latter. Then the genes identified as candidate can be subject to GO analysis to look for functional gene classes. 

Once we have candidate gene, there is a resequencing study. That gene and regulatory sequences are assayed in large populations - are there non-synonymous changes? Was the non-synonymous SNP and outlier within the gene?

Use of Crispr-cas9 into cell lines of wild populations to understand genetic/cellular consequences. seems cool but potentially limited- how can we really extend from the impacts in a cell line to full organisms? 



