---
title: "Genetic Drift"
author: "Gaurav Kandlikar"
date: "November 20, 2015"
output: pdf_document
---

```{r echo = F}
library(wesanderson)
```

### Question 1

\textsc{What is the expected number of successes in a sample of size 10 from the binomial distribution with probability of success p=0.1?}    
First, figure this out analytically based on the formulas from class. Second, write a simulation in R to confirm this.

### Response to 1   

$E[X] = np = 0.1 * 10 = 1$ 

```{r question-1, cache=T}
reps <- 10000
n <- 10
p <- 0.1

table(rbinom(reps, n, p))
```

The simulations above show that the expected value of this is 1.


### Using R to simulate genetic drift

Consider that `p` is the allele frequency in the population; `N` is the number of diploid individuals. 

```{r genetic-drift-1, cache=T}
p <- 0.1
N <- 10
count <- rbinom(1, 2*N, p)
count/(2*N)
```


### Question 2
\textsc{Write a function in R that will simulate T generations of genetic drift for L independent SNPs. Keep track of the allele frequencies of each of the L SNPs in each of the T generations. All SNPs should start in the initial generation at frequency p.}

```{r question-2, cache=T}
gene_frequencies <- function(num_generations = 10, init_freq = 0.1, num_genes = 3, pop_size = 100) {
  
  full_matrix <- matrix(nrow = num_generations, ncol = num_genes)
  full_matrix[1, ] <- rep(init_freq, num_genes)
  for (i in 2:num_generations) {
    for (j in 1:num_genes) {
      freq <- rbinom(1, 2*pop_size, full_matrix[i-1, j])/(2*pop_size)
      full_matrix[i,j] <- freq
    }
  }
  
  return(full_matrix)
  
}

freqs <- gene_frequencies(num_generations = 10000, init_freq = 0.1, num_genes = 1000, pop_size = 100)

# How many of the 1000 SNP are at frequency 0/1 at the end of the simulation (in generation 10000)?
table(freqs[nrow(freqs),])
```

\textsc{Does this value agree with the theoretical prediction for the probability of fixation of a neutral allele?}
As shown above, ~900 of the SNPs were at frequency of zero at the end of the simulation; ~100 were at fixation. This lines up with expectation, which is that the likelihood of fixation is equal to the initial frequency of a neutral SNP. 

```{r question-2-, cache=T}
cols <- wes_palette("Darjeeling", n = 1000, "continuous")
plot(freqs[,1], type = "l", ylim = c(-0.05, 1.05), xlab = "generation", ylab = "allele frequency",
     main = "Trajectory of allele frequency", col  = cols[1])
invisible(sapply(2:ncol(freqs), function(x) lines(freqs[,x], col = cols[x])))
```

\textsc{Repeat the simulation, but this time set p=0.6} 

```{r, cache=T}
freqs <- gene_frequencies(num_generations = 10000, init_freq = 0.6, num_genes = 1000, pop_size = 100)

# How many of the 1000 SNP are at frequency 0/1 at the end of the simulation (in generation 10000)? 
table(freqs[nrow(freqs),])
```

\textsc{Does this value agree with the theoretical prediction for the probability of fixation of a neutral allele?}  
Yes- the probability of fixation of a neutral allele is equal to its initial frequency in the population, which in this case is 0.6.

### Question 4   

\textsc{Let’s look at the effect of the population size on patterns of genetic drift. Repeat the simulation, but this time, set N=10, 500 and 1000.  Keep the other parameters the same (p=0.1; L=1000; T=10000). Again, N is the number of diploids.}

```{r fig.height=8, fig.width=10, cache=T}
pop_sizes <- c(10, 100, 500, 1000)
# pop_sizes <- c(10)
frequencies_by_size <- lapply(pop_sizes, function(x)
  gene_frequencies(num_generations = 10000, init_freq = 0.1, num_genes = 1000, pop_size = x))

str(frequencies_by_size)
names(frequencies_by_size) <- pop_sizes
cols <- wes_palette("Darjeeling", n = 100, "continuous")
par(mfrow = c(2,2))

for(i in 1:4){
  plot(frequencies_by_size[[i]][,1], type = "l", ylim = c(-0.05, 1.05), xlab = "generation", 
       ylab = "allele frequency", main = "Trajectory of allele frequency", col  = cols[1])
  sapply((2:100), function(x) lines(frequencies_by_size[[i]][,x], col = cols[x]))
}

```

\textsc{Based on examination of the plots, how does the population size affect allele frequency change?}     

Alleles seem to become fixed or extinct in a small population much faster than in a large population. 

\textsc{For each population size, in what proportion of simulation replicates did the derived allele become fixed by the end of the simulation?}

```{r question-4-d, cache = T}
prop_fixed <- function(x) {
  vv <- x[nrow(x), ]
  rr <- sum(vv == 1)/ncol(x)
  return(rr)
}
sapply(frequencies_by_size, prop_fixed)
```

As seen above, the proportion of replicates that went to fixation was about the expected value of 0.1 regardless of the population size. Another interesting way to explore this question would be plot the proportion of fixed sites as a function of time- I would expect that alleles go to fixation much faster in the small populations:

```{r prop-alleles-fixed-v-time, cache = T}
prop_fixed_by_time <- function(x){
  
  vv <- numeric(nrow(x))
  for (i in 1:nrow(x)) {
    qq <- x[i, ]
    vv[i] <- sum(qq == 1)/ncol(x)
  }
  return(vv)
}

fixed_alleles <- sapply(frequencies_by_size, prop_fixed_by_time)
cols <- wes_palette("Darjeeling", n = 4)
par(mfrow = c(1,1))
plot(fixed_alleles[,1], type = "l", col = cols[1], lwd = 2,
     main = "Proportion of alleles fixed over time", ylab = "Prop alleles", xlab = "generation")
for(i in 2:4) {lines(fixed_alleles[,i], col = cols[i], lwd = 2)}
legend("bottomright", lty = 1, col = cols, legend = c("N = 10", "N = 100", "N = 500", "N = 1000"),
       bty = "n", lwd = 2)
```


\textsc{How does this probability of fixation estimated from the simulations match with the theoretical prediction?}    

Even though the expected value of $p$ is equal regardless of the population size, theory tells us that the variance in $p$ does indeed depend on $N$. In other words,  

$$E[p_{t+1}] = p_{t}$$,
$$V[p_{t+1}] = \frac{p_{t}*(1-p_{t})}{2N}$$

The equation for variance shows that the variance i $p$ decreases with population size, so we expect the allele frequencies to show more drastic shifts in small populations than in large ones. 


## Coalescence

### Exploring the exponential distribution


```{r exponential-distribution, cache = T}
# E[X] in exponential distribution == 1/rate
rate = c(5,10)
1/rate; sapply(rate, function(x) mean(rexp(n = 100, rate = x)))
```

### Question 5

\textsc{What is the rate of coalescence for a sample size of 2 chromosomes in a population of size 2N chromosomes? This is akin to the probability that 2 chromosomes find a common ancestor in the previous generation.}

The rate of coalescence for two chromosomes in a population of size 2N is $\frac{1}{2N}$

### Question 6   

\textsc{Perform 10,000 simulations of coalescent times for a sample size of 2 chromosomes from this population of size 2N=20,000.} 
```{r question-6-a, cache = T}
reps <- 10000
N <- 10000

rate <- 1/((2*N))
coal_times <- rexp(reps, rate)
# for(i in 1:length(coal_times)) {
# coal_times[i] <- mean(rexp(2, rate))
# }

2*N ; mean(coal_times)
```

\textsc{What is the average time to the most recent common ancestor (TMRCA) in your simulations? b) What is the theoretical expectation? c) How do the two values compare?}

As shown above, the average time to MRCA in the simulations is ~20000. The theoretical expectation for coalescence time is $2N$ = 20000 generations. 

\textsc{Make a density plot of your simulated coalescent times.}
```{r question-6-d}
plot(density(coal_times), main = "Distribution of TMRCAs", xlab = "generations")
```

\textsc{What is the standard deviation of the coalescent times?}
```{r question-6-e}
sd(coal_times)
```

The standard deviation of this is also equal to ~20000. This makes sense, as the sd of an exponential distribution is, like its mean, equal to $1/rate$. 

### Question 7   
\textsc{Repeat the simulation described in question 6, but this time set 2N=2,000. Answer parts a-e from question 6 for this new set of simulations. }

```{r question-7-a, cache = T}
reps <- 10000
N <- 1000

rate <- 1/((2*N))
coal_times <- rexp(reps, rate)

2*N ; mean(coal_times); sd(coal_times)
plot(density(coal_times), main = "Distribution of TMRCAs (2N = 2000)",
     xlab = "generations")

```

\textsc{How does the average TMRCA from the simulations in question 6 (where 2N=20,000) compare to the average TMRCA from the simulations in question 7 (2N=2,000)}  

The TMRCA in the 2N = 2000 population is ~2000 generations, while it is ~20000 generations in the 2N = 20000 population. 

\textsc{What can you conclude about how the population size affects the expected coalescent time? }  

This exercise shows that the expected time to coalescent grows linearly with the number of chromosomes in the population. 

### Question 8  

\textsc{Let’s add mutations to the genealogies you’ve simulated and then look at what these models predict in terms of the genetic variation you might see in a sample. Set mu = 1 x 10-4}

\textsc{Add mutations to the genealogies (really the coalescent times) that you simulated in question 6 (N=20,000) using the algorithm I just outlined}
```{r question-8-a, cache = T}
# Regenerate coal_times with 2N = 20000
reps <- 10000
N <- 10000

rate <- 1/((2*N))
coal_times <- rexp(reps, rate)

# mean(coal_times)

mu <- 1*10^-4
num_snps <- numeric(reps)
for (i in 1:length(coal_times)) {
  net_mutation_rate <- 2*mu*coal_times[i]
  num_snps[i] <- rpois(1, net_mutation_rate)
}
```

\textsc{What is the average number of SNPs per geneology}    

```{r, cache = T}
mean(num_snps)
```

\textsc{Recall that $\theta = 4N\mu$. Note, the factor-of-2-confusion—N refers to the number of diploids. So, $\theta$ is twice the number of chromosomes multiplied by mu. What is $\theta$ predicted to be in this example?}

```{r, cache = T}
# Recall mu and N
mu; N

# Twice the number of chromosomes multiplied by mu
theta <- 4*N*mu; theta; mean(num_snps)
```

\textsc{Pretty neat, huh?}

Sure is!

\textsc{Make a density plot of the number of SNPs per simulation replicate.}

```{r}
plot(density(num_snps), main = "Distribution of SNP count",
     xlab = "SNP count")
legend("topright", legend = c(bquote(mu == .(mu)), bquote(N == .(N))), bty = "n")
```

### Question 10 (sic)   
\textsc{Repeat all the parts in question 8, but this time, set N=2,000.}

```{r question-10}

# Regenerate coal_times with 2N = 2000
reps <- 10000
N <- 1000

rate <- 1/((2*N))
coal_times_small <- rexp(reps, rate)

mean(coal_times)

mu <- 1*10^-4
num_snps_small <- numeric(reps)
for (i in 1:length(coal_times_small)) {
  net_mutation_rate <- 2*mu*coal_times_small[i]
  num_snps_small[i] <- rpois(1, net_mutation_rate)
}

mean(num_snps_small)
theta <- 4*N*mu; theta; mean(num_snps_small)

plot(density(num_snps_small), main = "Distribution of SNP count",
     xlab = "SNP count")
legend("topright", legend = c(bquote(mu == .(mu)), bquote(N == .(N))), bty = "n")
```


```{r question-11, cache = T}
par(mfrow = c(1,2))
plot(x = coal_times_small, y = num_snps_small, main = "SNPs v TMRCA", 
     xlab = "Time to coalescence", ylab = "Number of SNPs accumulated")
mtext(side = 3, text = "N = 1000")
plot(x = coal_times, y = num_snps, main = "SNPs v TMRCA", 
     xlab = "Time to coalescence", ylab = "Number of SNPs accumulated")
mtext(side = 3, text = "N = 10000")
```

\textsc{How are these variables related to each other?}  

Broadly, it appears that at sufficiently large population sizes, the number of SNPs is significantly correlated to the TMRCA. The linear relationship seems to get stronger with population size- below are the R^2 values for the two population size scenarios (I understand that I am breaking assumptions of normally distributed data, etc., given that this is count data... but this seems to be a decent first-cut?).  

```{r}
rsq_small_pop <- summary(lm(coal_times_small~num_snps_small))$adj.r.squared
rsq_big_pop <- summary(lm(coal_times~num_snps))$adj.r.squared

round(rsq_small_pop, 4); round(rsq_big_pop, 4)
```

\textsc{Does the number of SNPs tell you anything about the TMRCA?}

If we have an estimate of SNP count in a population and a reliable model to relate SNPs to TMRCAs (having a good model requires a good estimate of mutation rate, which may not be readily available ubiquitously), we could reconstruct the demographic history of a population.  

### Question 12   

\textsc{Imagine that you sequenced a 10kb region of DNA from single individual (a diploid species; i.e. you have a sample of 2 chromosomes).  You observed 10 SNPs in that 10kb interval. Note that independent evidence suggests a mutation rate of 1 x 10-8 per base pair/generation. (Hint: this is very similar to what you’ve simulated)}

\textsc{Use coalescent simulations to evaluate whether a model with N=10,000 is compatible with your observed data? Put another way, what proportion of simulation replicates from this model have >= 10 SNPs?}

```{r}
N <- 10000
mu <- 1*10^-8
bps <- 10000

# Step 1: simulate times to coalescent ---------
reps <- 10000
rate <- 1/(2*N)

coal_times_12 <- rexp(reps, rate)
# Mean of simulation; Expected
mean(coal_times_12); 2*N
# SD of simulation; Expected
sd(coal_times_12); 2*N

# Step 2: simulate SNPs ------------

num_snps_12 <- numeric(reps)

for(i in 1:length(coal_times_12)) {
  net_mutation_rate <- 2*bps*mu*coal_times_12[i]
  num_snps_12[i] <- rpois(1, net_mutation_rate)
}
mean(num_snps_12)
4*N*mu*bps

# Plot
par(mfrow = c(1,1))
plot(density(num_snps_12), main = "Density of SNPs", xlab = "Num SNPs")
abline(v = 10, lwd = 2)

# Proportion of simulations compatible 
sum(num_snps_12 >= 10)/length(num_snps_12)
```